packages/web/src/common/store/cache/collections/commonSagas.ts

```diff
@@ -467,75 +509,80 @@ function* confirmPublishPlaylist(
         }
 
-        yield put(toast({ content: 'Your playlist is now public!' }))
+        yield* put(toast({ content: 'Your playlist is now public!' }))
       },
       function* ({ error, timeout, message }) {
         // Fail Call
-        yield put(
+        yield* put(
           collectionActions.publishPlaylistFailed(
-            message,
+            error,
             { userId, playlistId },
             { error, timeout }
           )
         )
       },
-      (result) => (result.playlist_id ? result.playlist_id : playlistId)
+      (result: Collection) =>
+        result.playlist_id ? result.playlist_id : playlistId
     )
   )
 }
 
 /** DELETE PLAYLIST */
 
 function* watchDeletePlaylist() {
-  yield takeEvery(collectionActions.DELETE_PLAYLIST, deletePlaylistAsync)
+  yield* takeEvery(collectionActions.DELETE_PLAYLIST, deletePlaylistAsync)
 }
 
-function* deletePlaylistAsync(action) {
-  yield waitForWrite()
-  const userId = yield select(getUserId)
+function* deletePlaylistAsync(
+  action: ReturnType<typeof collectionActions.deletePlaylist>
+) {
+  yield* waitForWrite()
+  const userId = yield* select(getUserId)
   if (!userId) {
-    yield put(signOnActions.openSignOn(false))
+    yield* put(signOnActions.openSignOn(false))
     return
   }
 
   // Depending on whether the collection is an album
   // or playlist, we should either delete all the tracks
   // or just delete the collection.
-  const collection = yield select(getCollection, { id: action.playlistId })
+  const collection = yield* select(getCollection, { id: action.playlistId })
   if (!collection) return
 
   const isAlbum = collection.is_album
   if (isAlbum) {
     const trackIds = collection.playlist_contents.track_ids
 
     const event = make(Name.DELETE, { kind: 'album', id: action.playlistId })
-    yield put(event)
-    yield call(confirmDeleteAlbum, action.playlistId, trackIds, userId)
+    yield* put(event)
+    confirmDeleteAlbum(action.playlistId, trackIds, userId)
+    yield* call(confirmDeleteAlbum, action.playlistId, trackIds, userId)
   } else {
     const event = make(Name.DELETE, { kind: 'playlist', id: action.playlistId })
-    yield put(event)
+    yield* put(event)
 
     // Preemptively mark the playlist as deleted.
     // It's possible there are other transactions confirming
     // for this playlist, which prevent the delete confirmation
     // from running immediately, which would leave
     // the playlist visible before it runs.
-    yield put(
+    yield* put(
       cacheActions.update(Kind.COLLECTIONS, [
         {
           id: action.playlistId,
           metadata: { _marked_deleted: true }
         }
       ])
     )
-    yield call(confirmDeletePlaylist, userId, action.playlistId)
+    yield* call(confirmDeletePlaylist, userId, action.playlistId)
   }
 
-  const user = yield select(getUser, { id: userId })
-  yield put(
+  const user = yield* select(getUser, { id: userId })
+  if (!user) return
```

Comment:
thinking about whether we should return here or dispatch failed action

Reply:
probably ok to just return here. User will almost always be here and if its not, there shouldn't be any visible bugs by skipping this cache update

Reply:
This type of case comes up a lot, and I want to make sure we don't break anything. But yeah this one is likely fine 👍